### PR TITLE
Add list devices

### LINKED ISSUE #3: Add list devices
As a platform user, I want to list the devices registered in the fog.

- [ ] Add event listener to list devices
- [ ] List devices from the things service
- [ ] Send device list to the connector
- [ ] Update documentation

### PR DESCRIPTION
**Pull request by netoax**
What does this implement/fix?
---------------------------------------------------
This patch adds list things command capability.

Does this close any currently open issues?
------------------------------------------
Closes #3

### CODE REVIEW

File: pkg/thing/mocks/thing_proxy.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package mocks
+
+import (
+	"github.com/CESARBR/knot-babeltower/pkg/thing/delivery/http"
+	"github.com/CESARBR/knot-babeltower/pkg/thing/entities"
+	"github.com/stretchr/testify/mock"
+)
+
+type FakeThingProxy struct {
```
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported type FakeThingProxy should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 19
           

Complexity increasing per file
==============================
- pkg/thing/interactors/list_things.go  4
- pkg/thing/mocks/publisher.go  1
- pkg/thing/mocks/thing_proxy.go  1
- pkg/thing/interactors/update_schema_test.go  3
- pkg/thing/mocks/logger.go  1
- pkg/thing/interactors/update_schema.go  8
- pkg/thing/handler/amqp/msg_handler.go  5
- pkg/thing/interactors/list_things_test.go  5
- pkg/thing/mocks/connector.go  1
         

Complexity decreasing per file
==============================
+ pkg/thing/delivery/http/thing_proxy.go  -2
         

Clones added
============
- pkg/thing/interactors/update_schema_test.go  13
- pkg/thing/interactors/list_things_test.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780&bid=16102946)

File: pkg/thing/mocks/publisher.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package mocks
+
+import (
+	"github.com/CESARBR/knot-babeltower/pkg/network"
+	"github.com/CESARBR/knot-babeltower/pkg/thing/entities"
+	"github.com/stretchr/testify/mock"
+)
+
+type FakePublisher struct {
```
Comment (joaoaneto) [inline pkg/thing/mocks/publisher.go]: Codacy Issue found: [exported type FakePublisher should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+func (fl *FakeLogger) Infof(string, ...interface{}) {}
+
+func (fl *FakeLogger) Debug(...interface{}) {}
+
+func (fl *FakeLogger) Warn(...interface{}) {}
+
+func (fl *FakeLogger) Error(...interface{}) {}
+
+func (fl *FakeLogger) Errorf(string, ...interface{}) {}
```
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Errorf should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/publisher.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	ret := fp.Called(msg)
+	return ret.Error(0)
+}
+
+func (fp *FakePublisher) SendUpdatedSchema(thingID string) error {
+	ret := fp.Called(thingID)
+	return ret.Error(0)
+}
+
+func (fp *FakePublisher) SendThings(things []*entities.Thing) error {
```
Comment (joaoaneto) [inline pkg/thing/mocks/publisher.go]: Codacy Issue found: [exported method FakePublisher.SendThings should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package mocks
+
+type FakeLogger struct{}
```
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported type FakeLogger should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+func (fl *FakeLogger) Info(...interface{}) {}
+
+func (fl *FakeLogger) Infof(string, ...interface{}) {}
+
+func (fl *FakeLogger) Debug(...interface{}) {}
```
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Debug should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/connector.go [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+	SendError error
+	RecvError error
+}
+
+func (fc *FakeConnector) SendRegisterDevice(id, name string) (err error) {
+	ret := fc.Called(id, name)
+	return ret.Error(0)
+}
+
+func (fc *FakeConnector) SendUpdateSchema(id string, schemaList []entities.Schema) (err error) {
```
Comment (joaoaneto) [inline pkg/thing/mocks/connector.go]: Codacy Issue found: [exported method FakeConnector.SendUpdateSchema should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/connector.go [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+	"github.com/stretchr/testify/mock"
+)
+
+type FakeConnector struct {
+	mock.Mock
+	SendError error
+	RecvError error
+}
+
+func (fc *FakeConnector) SendRegisterDevice(id, name string) (err error) {
```
Comment (joaoaneto) [inline pkg/thing/mocks/connector.go]: Codacy Issue found: [exported method FakeConnector.SendRegisterDevice should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+func (fl *FakeLogger) Warn(...interface{}) {}
```
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Warn should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go [hunk 10]
```diff
@@ -0,0 +1,15 @@
+package mocks
+
+type FakeLogger struct{}
+
+func (fl *FakeLogger) Info(...interface{}) {}
```
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Info should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/publisher.go [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+	SendError error
+	Token     string
+}
+
+func (fp *FakePublisher) SendRegisterDevice(msg network.RegisterResponseMsg) error {
+	ret := fp.Called(msg)
+	return ret.Error(0)
+}
+
+func (fp *FakePublisher) SendUpdatedSchema(thingID string) error {
```
Comment (joaoaneto) [inline pkg/thing/mocks/publisher.go]: Codacy Issue found: [exported method FakePublisher.SendUpdatedSchema should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/thing_proxy.go [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+	ret := ftp.Called(id, name, authorization)
+	return ret.String(0), ret.Error(1)
+}
+
+func (ftp *FakeThingProxy) UpdateSchema(authorization, thingID string, schema []entities.Schema) error {
+	ret := ftp.Called(thingID, schema)
+	return ret.Error(0)
+}
+
+func (ftp *FakeThingProxy) Get(authorization, thingID string) (*http.ThingProxyRepr, error) {
```
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported method FakeThingProxy.Get should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/publisher.go [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+)
+
+type FakePublisher struct {
+	mock.Mock
+	ReturnErr error
+	SendError error
+	Token     string
+}
+
+func (fp *FakePublisher) SendRegisterDevice(msg network.RegisterResponseMsg) error {
```
Comment (joaoaneto) [inline pkg/thing/mocks/publisher.go]: Codacy Issue found: [exported method FakePublisher.SendRegisterDevice should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+
+func (fl *FakeLogger) Infof(string, ...interface{}) {}
```
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Infof should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/thing_proxy.go [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+
+	return nil, nil
+}
+
+func (ftp *FakeThingProxy) List(authorization string) ([]*entities.Thing, error) {
```
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported method FakeThingProxy.List should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/connector.go [hunk 16]
```diff
@@ -0,0 +1,27 @@
+package mocks
+
+import (
+	"github.com/CESARBR/knot-babeltower/pkg/thing/entities"
+	"github.com/stretchr/testify/mock"
+)
+
+type FakeConnector struct {
```
Comment (joaoaneto) [inline pkg/thing/mocks/connector.go]: Codacy Issue found: [exported type FakeConnector should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/connector.go [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+	ret := fc.Called(id, name)
+	return ret.Error(0)
+}
+
+func (fc *FakeConnector) SendUpdateSchema(id string, schemaList []entities.Schema) (err error) {
+	ret := fc.Called(id, schemaList)
+	return ret.Error(0)
+}
+
+func (fc *FakeConnector) RecvRegisterDevice() (bytes []byte, err error) {
```
Comment (joaoaneto) [inline pkg/thing/mocks/connector.go]: Codacy Issue found: [exported method FakeConnector.RecvRegisterDevice should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/thing_proxy.go [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+	"github.com/CESARBR/knot-babeltower/pkg/thing/entities"
+	"github.com/stretchr/testify/mock"
+)
+
+type FakeThingProxy struct {
+	mock.Mock
+	ReturnErr error
+}
+
+func (ftp *FakeThingProxy) Create(id, name, authorization string) (idGenerated string, err error) {
```
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported method FakeThingProxy.Create should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/thing_proxy.go [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+	ret := ftp.Called(id, name, authorization)
+	return ret.String(0), ret.Error(1)
+}
+
+func (ftp *FakeThingProxy) UpdateSchema(authorization, thingID string, schema []entities.Schema) error {
```
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported method FakeThingProxy.UpdateSchema should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 19
           

Complexity increasing per file
==============================
- pkg/thing/mocks/logger.go  1
- pkg/thing/interactors/list_things_test.go  5
- pkg/thing/mocks/connector.go  1
- pkg/thing/interactors/update_schema_test.go  3
- pkg/thing/handler/amqp/msg_handler.go  5
- pkg/thing/mocks/publisher.go  1
- pkg/thing/interactors/update_schema.go  8
- pkg/thing/interactors/list_things.go  4
- pkg/thing/mocks/thing_proxy.go  1
         

Complexity decreasing per file
==============================
+ pkg/thing/delivery/http/thing_proxy.go  -2
         

Clones added
============
- pkg/thing/interactors/list_things_test.go  2
- pkg/thing/interactors/update_schema_test.go  13
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780&bid=16102946)

File: pkg/thing/mocks/thing_proxy.go (hunk 15 above)
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported method FakeThingProxy.List should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go (hunk 6 above)
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Debug should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/connector.go (hunk 7 above)
Comment (joaoaneto) [inline pkg/thing/mocks/connector.go]: Codacy Issue found: [exported method FakeConnector.SendUpdateSchema should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go (hunk 14 above)
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Infof should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/publisher.go (hunk 11 above)
Comment (joaoaneto) [inline pkg/thing/mocks/publisher.go]: Codacy Issue found: [exported method FakePublisher.SendUpdatedSchema should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go (hunk 10 above)
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Info should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/publisher.go (hunk 4 above)
Comment (joaoaneto) [inline pkg/thing/mocks/publisher.go]: Codacy Issue found: [exported method FakePublisher.SendThings should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go (hunk 5 above)
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported type FakeLogger should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/connector.go (hunk 17 above)
Comment (joaoaneto) [inline pkg/thing/mocks/connector.go]: Codacy Issue found: [exported method FakeConnector.RecvRegisterDevice should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go (hunk 3 above)
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Errorf should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/publisher.go (hunk 2 above)
Comment (joaoaneto) [inline pkg/thing/mocks/publisher.go]: Codacy Issue found: [exported type FakePublisher should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/thing_proxy.go (hunk 12 above)
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported method FakeThingProxy.Get should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go (hunk 9 above)
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Warn should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/thing_proxy.go (hunk 1 above)
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported type FakeThingProxy should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/connector.go (hunk 8 above)
Comment (joaoaneto) [inline pkg/thing/mocks/connector.go]: Codacy Issue found: [exported method FakeConnector.SendRegisterDevice should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/thing_proxy.go (hunk 19 above)
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported method FakeThingProxy.UpdateSchema should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/connector.go (hunk 16 above)
Comment (joaoaneto) [inline pkg/thing/mocks/connector.go]: Codacy Issue found: [exported type FakeConnector should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/publisher.go (hunk 13 above)
Comment (joaoaneto) [inline pkg/thing/mocks/publisher.go]: Codacy Issue found: [exported method FakePublisher.SendRegisterDevice should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/thing_proxy.go (hunk 18 above)
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported method FakeThingProxy.Create should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 19
           

Complexity increasing per file
==============================
- pkg/thing/interactors/list_things_test.go  5
- pkg/thing/mocks/connector.go  1
- pkg/thing/mocks/thing_proxy.go  1
- pkg/thing/interactors/update_schema_test.go  3
- pkg/thing/interactors/list_things.go  4
- pkg/thing/mocks/publisher.go  1
- pkg/thing/handler/amqp/msg_handler.go  4
- pkg/thing/mocks/logger.go  1
- pkg/thing/interactors/update_schema.go  8
         

Complexity decreasing per file
==============================
+ pkg/thing/delivery/http/thing_proxy.go  -2
         

Clones added
============
- pkg/thing/interactors/list_things_test.go  2
- pkg/thing/interactors/update_schema_test.go  13
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780&bid=16102946)

File: pkg/thing/mocks/publisher.go (hunk 13 above)
Comment (joaoaneto) [inline pkg/thing/mocks/publisher.go]: Codacy Issue found: [exported method FakePublisher.SendRegisterDevice should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/publisher.go (hunk 4 above)
Comment (joaoaneto) [inline pkg/thing/mocks/publisher.go]: Codacy Issue found: [exported method FakePublisher.SendThings should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go (hunk 14 above)
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Infof should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/connector.go (hunk 8 above)
Comment (joaoaneto) [inline pkg/thing/mocks/connector.go]: Codacy Issue found: [exported method FakeConnector.SendRegisterDevice should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/publisher.go (hunk 2 above)
Comment (joaoaneto) [inline pkg/thing/mocks/publisher.go]: Codacy Issue found: [exported type FakePublisher should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go (hunk 10 above)
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Info should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go (hunk 6 above)
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Debug should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go (hunk 3 above)
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Errorf should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/thing_proxy.go (hunk 12 above)
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported method FakeThingProxy.Get should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/thing_proxy.go (hunk 18 above)
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported method FakeThingProxy.Create should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/thing_proxy.go (hunk 19 above)
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported method FakeThingProxy.UpdateSchema should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go (hunk 9 above)
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported method FakeLogger.Warn should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/connector.go (hunk 16 above)
Comment (joaoaneto) [inline pkg/thing/mocks/connector.go]: Codacy Issue found: [exported type FakeConnector should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/connector.go (hunk 17 above)
Comment (joaoaneto) [inline pkg/thing/mocks/connector.go]: Codacy Issue found: [exported method FakeConnector.RecvRegisterDevice should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/thing_proxy.go (hunk 15 above)
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported method FakeThingProxy.List should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/thing_proxy.go (hunk 1 above)
Comment (joaoaneto) [inline pkg/thing/mocks/thing_proxy.go]: Codacy Issue found: [exported type FakeThingProxy should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/connector.go (hunk 7 above)
Comment (joaoaneto) [inline pkg/thing/mocks/connector.go]: Codacy Issue found: [exported method FakeConnector.SendUpdateSchema should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/publisher.go (hunk 11 above)
Comment (joaoaneto) [inline pkg/thing/mocks/publisher.go]: Codacy Issue found: [exported method FakePublisher.SendUpdatedSchema should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

File: pkg/thing/mocks/logger.go (hunk 5 above)
Comment (joaoaneto) [inline pkg/thing/mocks/logger.go]: Codacy Issue found: [exported type FakeLogger should have comment or be unexported](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/handler/amqp/msg_handler.go  2
- pkg/thing/mocks/publisher.go  1
- pkg/thing/interactors/list_things_test.go  5
- pkg/thing/mocks/connector.go  1
- pkg/thing/interactors/list_things.go  4
- pkg/thing/mocks/thing_proxy.go  1
- pkg/thing/mocks/logger.go  1
         

Clones added
============
- pkg/thing/interactors/list_things_test.go  2
- pkg/thing/interactors/update_schema_test.go  2
         

Clones removed
==============
+ pkg/thing/interactors/register_thing_test.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=4853780&bid=16102946)

File: pkg/thing/delivery/http/thing_proxy.go [hunk 20]
```diff
@@ -205,6 +206,20 @@ func (p proxy) UpdateSchema(authorization, ID string, schemaList []entities.Sche
 	return p.mapErrorFromStatusCode(resp.StatusCode)
 }
 
+func (p proxy) List(authorization string) (things []*entities.Thing, err error) {
+	pagThings, err := p.getPaginatedThings(authorization)
+	if err != nil {
+		return nil, nil
+	}
+
+	for i := range pagThings {
```
Comment (ramonhpr) [inline pkg/thing/delivery/http/thing_proxy.go]:
```suggestion
	for _, t := range pagThings {
```

File: pkg/thing/delivery/http/thing_proxy.go [hunk 21]
```diff
@@ -205,6 +206,20 @@ func (p proxy) UpdateSchema(authorization, ID string, schemaList []entities.Sche
...
+		t := pagThings[i]
```
Comment (ramonhpr) [inline pkg/thing/delivery/http/thing_proxy.go]:
```suggestion
```

**Review by ramonhpr — CHANGES_REQUESTED** [review]

File: pkg/thing/delivery/http/thing_proxy.go (hunk 20 above)
Comment (netoax) [reply pkg/thing/delivery/http/thing_proxy.go]: Done. Thanks!

File: pkg/thing/delivery/http/thing_proxy.go (hunk 21 above)
Comment (netoax) [reply pkg/thing/delivery/http/thing_proxy.go]: Done. Thanks!

**Review by ramonhpr — APPROVED** [review]

**Pull request merged**